### PR TITLE
Map Exporter Revamp + Cleanup

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
Ports most of BeeStation/BeeStation-Hornet#1522 with drastic changes/improvements/adaptations

## Why It's Good For The Game
Allows admins to more easily and efficiently export maps of cool ships and outposts, without lagging the entire server to heck.

## Changelog
:cl: PowerfulBacon, Mark Suckerberg
code: Removes old, deprecated map generator code
code: Removes unused, broken minimap code
refactor: Refactors map exporting code to export to TGM and not lag as much
fix: The `map exp` build mode module now actually works again 
add: Adds option selection to the `map exp` build mode module, accessible by right clicking it
del: Minimap code (was broken anyways)
del: Old procedural generation code (wasn't used anyways)
/:cl:

### CODE REVIEW

**Comment by shiptest-server** [comment]
#### Test Merge Deployed

##### Server Instance
shiptest

##### Revision
Origin: f789e74004b3acb82690d77398848b240a153b6f
Pull Request: 3916e830b854c73665e6a12a7ebe6169771f36f7
Server: 1dbd0bfb0d4198bd0a0d556e3729419b22e55678

**Comment by shiptest-server** [comment]
#### Test Merge Updated

##### Server Instance
shiptest

##### Revision
Origin: 89b653c96fc59fa415eeb26666f154fe97f257c1
Pull Request: 3916e830b854c73665e6a12a7ebe6169771f36f7
Server: 9172c1bfefbcd61b7e2dccb8f8cceb235072078e

**Comment by shiptest-server** [comment]
#### Test Merge Updated

##### Server Instance
shiptest

##### Revision
Origin: 7c3512bd559cf2b1f8e230216ff853c7f118d7ee
Pull Request: 3916e830b854c73665e6a12a7ebe6169771f36f7
Server: 7f432c12878b85f2e0187a8393c33b9d5b197880

**Comment by shiptest-server** [comment]
#### Test Merge Updated

##### Server Instance
shiptest

##### Revision
Origin: bf2a3d7daffd8042d3adbe075ec803cd3abd4e03
Pull Request: 3916e830b854c73665e6a12a7ebe6169771f36f7
Server: 68914e28c7ca55ffe87c21c0fc6dbcb9af418093

**Comment by shiptest-server** [comment]
#### Test Merge Removed

File: code/modules/mapping/writer.dm [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+* for .dmm files. This file can then be opened in the map editor or imported
+* back into the game.
+* ============================
+* This has been made semi-modular so you should be able to use these functions
+* elsewhere in code if you ever need to get a file in the .dmm format
+**/
+/atom/proc/get_save_vars()
+	return list("pixel_x", "pixel_y", "dir", "name", "req_access", "req_access_txt", "piping_layer", "color", "icon", "icon_state", "pipe_color", "amount")
+
+GLOBAL_LIST_INIT(save_file_chars, list(
```
Comment (ZephyrTFA) [inline code/modules/mapping/writer.dm:13]: Couldn't this be a regex instead? `[a-zA-Z]`

Comment (MarkSuckerberg) [reply code/modules/mapping/writer.dm:13]: Don't think so, these aren't used to find instances in which these are found, they're used to find the next char to use when generating a TGM key.

Comment (ZephyrTFA) [reply code/modules/mapping/writer.dm:13]: Ah, that makes sense. LGTM

**Pull request merged**